wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectManager.java

```diff
@@ -109,16 +107,22 @@ public ProjectManager(VirtualFileSystemProvider vfsProvider,
 
     @PostConstruct
     void initWatcher() throws IOException {
-        FileWatcherNotificationListener defaultListener = new FileWatcherNotificationListener(VirtualFileFilter.ACCEPT_ALL) {
-            @Override
-            public void onFileWatcherEvent(VirtualFile virtualFile, FileWatcherEventType eventType) {
-                LOG.debug("FS event detected: " + eventType + " " + virtualFile.getPath().toString() + " " + virtualFile.isFile());
-                eventService.publish(DtoFactory.newDto(VfsWatchEvent.class)
-                                               .withPath(virtualFile.getPath().toString())
-                                               .withFile(virtualFile.isFile())
-                                               .withType(eventType));
-            }
-        };
+        FileWatcherNotificationListener defaultListener =
+                new FileWatcherNotificationListener(file -> !(file.getPath().toString().contains(".codenvy")
```

Comment:
It is out of scope of this pull request, but indeed this whole section will be moved and rearranged to comply with dependency injection pattern with the help of Guice. That will basically allow us to easily configure filters for `FileTreeWatcher` for example.